src/extension/api.ts

```diff
@@ -14,4 +18,32 @@ async function getMergeRequestInfo(repoURLName,mergeRequestID) {
     }
 }
 
-export {getMergeRequestInfo}
\ No newline at end of file
+async function putRebaseRequest(repoURLName,mergeRequestID,setContentInDesc){
+    try{
+        const res = await ajaxClient.PUT(`projects/${encodeURIComponent(repoURLName)}/merge_requests/${mergeRequestID}/rebase`);
+        if (res.ok) {
+            setContentInDesc(`Rebase request is enqueued successfully`);
+            while(true){
+                try{
+                    const statusResponse = await ajaxClient.GET(`projects/${encodeURIComponent(repoURLName)}/merge_requests/${mergeRequestID}?include_rebase_in_progress=true`);
+                    const statusJSONresponse = await statusResponse.json();
+                    console.log(statusJSONresponse);
+                    if(statusJSONresponse.rebase_in_progress==false){
+                        return statusJSONresponse;
+                    }
+                    await wait(100);
+                }catch(e){
+                    console.log(e);
+                }
+            }
+        }
+        else {
+            throw new Error(`You are not authorized! Update your private token!`);
```

Comment:
Does the API return any error message ? If yes use that, otherwise add something like Rebase request failed. Please try again.

Reply:
Always use API Error messages when possible.